src/chat.cpp

```diff
@@ -22,38 +22,38 @@ bool PrivateChatChannel::isInvited(uint32_t guid) const
 
 bool PrivateChatChannel::removeInvite(uint32_t guid) { return invites.erase(guid) != 0; }
 
-void PrivateChatChannel::invitePlayer(const Player& player, Player& invitePlayer)
+void PrivateChatChannel::invitePlayer(std::shared_ptr<const Player> player, std::shared_ptr<Player> invitePlayer)
```

Comment:
I reviewed all the places that previously used `T&` or `const T&` and changed them to `std::shared_ptr<T>` or `std::shared_ptr<const T>`.  Looks good!😏